data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/buffer/BlockingBuffer.java

```diff
@@ -140,8 +146,8 @@ public static PluginSetting getDefaultPluginSettings() {
     }
 
     @Override
-    public void checkpoint(final CheckpointState checkpointState) {
-        final int numCheckedRecords = checkpointState.getNumCheckedRecords();
+    public void checkpoint() {
+        final int numCheckedRecords = numRecordsToBeChecked.getAndSet(0);
```

Comment:
I am not sure how this will work. numRecordsToBeChecked is an attribute of the BlockingBuffer, without the state you can not decrement or set to zero.

An example,

Lets assume the buffer size is 100, batch size 50 and we have two processors (aka processworker threads)

1. processor-1  reads 50 records, numRecordsToBeChecked is set to 50.
2. processor-2  reads 50 records,  numRecordsToBeChecked is set to 50.
3. processor-1 checkpoints, numRecordsToBecheckec is set to 0.

So, processor-2 is still working, but the buffer is read to accept more records.

Reply:
Thanks for the crucial catch. Now this PR breaks the previously desired functionally with checkpoint API.